### PR TITLE
[JBPM-10142] Improving clean up

### PR DESCRIPTION
**Pull request by fjtirado**
**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[link](https://www.example.com)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java [hunk 1]
```diff
@@ -235,34 +235,30 @@ public void disposeRuntimeEngine(RuntimeEngine runtime) {
...
-            	releaseAndCleanLock(runtime);
-        	}
+                }
+            }
     	} catch (Exception e) {
-    	    releaseAndCleanLock(runtime);
-    	    removeLocalRuntime(runtime);  	    
     	    throw new RuntimeException(e);
+    	} finally {
+    	    removeLocalRuntime(runtime);
```
Comment (gmunozfe) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java:260]: Shouldn't be checked `if (canDispose(runtime))` before?

Comment (fjtirado) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java:260]: removeLocalRuntimer should be invoked always, it is not related with dispose (you can check that in the previous version)

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by gmunozfe — APPROVED** [review]
Looks good to me, fantastic work @fjtirado !

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java (hunk 1 above)
Comment (fjtirado) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManager.java]: @gmunozfe you were right and we need to preserve existing order for current test to work

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**